### DISCUSSION TITLE
SoftGPU - VirtualBox/QEMU graphics device

### DISCUSSION THREAD

**Discussion by Torinde** [86Box/86Box#4573] [General]
As mentioned [here](https://github.com/JHRobotics/softgpu/issues/13#issuecomment-1675299897) it will be useful to support SoftGPU.

Currently it requires one of the [following devices](https://github.com/JHRobotics/vmdisp9x?tab=readme-ov-file#vmdisp9x):
- [x] Bochs VBE Extensions (Bochs: [VBE](https://github.com/86Box/86Box/discussions/2910#discussioncomment-10490036), VirtulBox: VboxVGA, QEMU: std-vga)
- [ ] VMWare SVGA-II (VMWare Workstation/Player, VirtulBox: VMSVGA, QEMU: vmware-svga)
- [ ] VBox SVGA (VirtulBox: VBoxSVGA)

Of which, hardware 3D acceleration is possible with the latter two.

- [ ] [Potentially relevant](https://github.com/JHRobotics/softgpu/issues/13#issuecomment-1628424754) would be to provide [QEMU virtio](https://www.kraxel.org/blog/2019/09/display-devices-in-qemu/) (also for [this driver](https://github.com/virtio-win/kvm-guest-drivers-windows/pull/943)).

Also possible is to [add support](https://github.com/JHRobotics/softgpu/issues/22) to an already emulated device (even if only its 2D part works currently) if there is an open source driver for it. The DirectX level depends on the DDI of that device (DX7 runs on any device driver interface/DDI, DX8 on DDI6+, DX9 on DDI7+)
-  DX8 base driver from emulation of Nvidia, ATI, Intel, S3/VIA, PowerVR, SiS, Trident, 3DLabs, Matrox
-  DX9 base driver from emulation of Nvidia, ATI, Intel, S3/VIA, PowerVR, SiS/XGI, 3DLabs

Related: #4144, #2910

**Comment by Mrtechenthuiast** [comment]
I'm sure there is some documentation of those video adapters out there...
